### ISSUE TITLE
defineSlots 如果没写 () 不报错

### ISSUE THREAD

**Issue by WangJincheng4869** [vuejs/eslint-plugin-vue#2295]
**Please describe what the rule should do:**

`defineSlots` 语法错误

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[x] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue

```

**Additional context**

![image](https://github.com/vuejs/language-tools/assets/41162158/2a46c517-f83f-4149-b52b-a1d9fc26dd7a)
正确写法：
```ts
defineSlots<{
  header(): any;
  main(): any;
}>();
```
错误写法：
```ts
defineSlots<{
  header(): any;
  main(): any;
}>;
```

**Comment by FloEdelmann** [comment]
Please write your issues in English, it's difficult to understand this suggestion. If I understood correctly, you are suggesting a new rule that warns about `defineSlots` if the parentheses `()` are missing.

@ota-meshi Do you think this rule is a good idea? Shouldn't the Vue compiler warn about this already?

**Comment by FloEdelmann** [comment]
Maybe it's also implicitly implemented by https://github.com/vuejs/eslint-plugin-vue/issues/2294, since `defineSlots` without `()` would not silence the warning about a used slot.

**Comment by WangJincheng4869** [comment]
> Please write your issues in English, it's difficult to understand this suggestion. If I understood correctly, you are suggesting a new rule that warns about `defineSlots` if the parentheses `()` are missing.
> 
> @ota-meshi Do you think this rule is a good idea? Shouldn't the Vue compiler warn about this already?

是的！您理解的是对的！（Yes! What you understand is correct!）

**Comment by ota-meshi** [comment]
I don't understand what you are requesting in this issue. So I'll close this issue.
Please write the issue in English and enter all necessary information.
